src/main/java/com/medallia/word2vec/neuralnetwork/NeuralNetworkTrainer.java

```diff
@@ -151,28 +156,30 @@ static long incrementRandom(long r) {
 	}
 	
 	/** @return Trained NN model */
-	public NeuralNetworkModel train(Iterable<List<String>> sentences) throws InterruptedException {
-		ListeningExecutorService ex = MoreExecutors.listeningDecorator(Executors.newFixedThreadPool(config.numThreads));
-		
+	public NeuralNetworkModel train(final Iterable<List<String>> sentences) throws InterruptedException {
+		final ListeningExecutorService ex =
+				MoreExecutors.listeningDecorator(
+					new ThreadPoolExecutor(config.numThreads, config.numThreads,
```

Comment:
Actually, that's not correct. The queue size matters when the main thread is running a task due to the `CallerRunsPolicy`. So it is connected to the number of processors. I changed it.